python/cli/prelude_cli/views/build.py

```diff
@@ -140,20 +141,38 @@ def upload(p: Path):
 @click.option('-s', '--source', help='source of threat (ex. www.cisa.gov)', default=None, type=str)
 @click.option('-i', '--source_id', help='ID of the threat, per the source (ex. aa23-075a)', default=None, type=str)
 @click.option('-t', '--tests', help='comma-separated list of test IDs', default=None, type=str)
+@click.option('-d', '--directory', help='directory containing tests and IOAs', default=None, type=click.Path(exists=True, dir_okay=True, file_okay=False))
 @click.pass_obj
 @handle_api_error
-def create_threat(controller, name, published, id, source_id, source, tests):
+def create_threat(controller, name, published, id, source_id, source, tests, directory):
     """ Create a security threat """
     with Spinner(description='Creating new threat'):
-        t = controller.create_threat(
-            name=name,
-            threat_id=id,
-            source_id=source_id,
-            source=source,
-            published=published,
-            tests=tests
-        )
-    print_json(data=t)
+        try:
+            created_tests = list()
+            uploads = list()
+            threat = None
+            if directory:
+                for technique_dir in os.listdir(directory):
+                    with open(f'{directory}/{technique_dir}/test.go', 'r') as f:
+                        go_code = f.read()
+                    with open(f'{directory}/{technique_dir}/config.json', 'r') as f:
+                        config = json.load(f)
+                    test = controller.create_test(name=config['name'], unit=config['unit'], technique=config['technique'])
+                    created_tests.append(test)
+                    uploads.append(controller.upload(test_id=test['id'], filename='test.go', data=go_code.encode()))
```

Comment:
nit: make the filename `<test_id>.go` (just so it matches what Prelude does internally and also what the CLI does if you do create_test / upload_attachment flow)